braindecode/augmentation/_functionals.py

```diff
@@ -0,0 +1,744 @@
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+
+    Parameters
+    ----------
+    x: torch.tensor, optional
+        Single EEG channel signal in time space. Should not be passed if f is
+        given. Defaults to None.
+    f: torch.tensor, optional
+        Fourier spectrum of a single EEG channel signal. Should not be passed
+        if x is given. Defaults to None.
+    eps: float, optional
+        Float between 0 and 1 setting the range over which the phase
+        pertubation is uniformly sampled: [0, `eps` * 2 * `pi`]. Defaults to 1.
+    random_state: int | numpy.random.Generator, optional
+        By default None.
+
+    References
+    ----------
+    .. [1] Schwabedal, J. T., Snyder, J. C., Cakmak, A., Nemati, S., &
+       Clifford, G. D. (2018). Addressing Class Imbalance in Classification
+       Problems of Noisy Signals by using Fourier Transform Surrogates. arXiv
+       preprint arXiv:1806.08675.
+    """
+    assert isinstance(
+        eps,
+        (Real, torch.FloatTensor, torch.cuda.FloatTensor)
+    ) and 0 <= eps <= 1, f"eps must be a float beween 0 and 1. Got {eps}."
+    if f is None:
+        assert x is not None, 'Neither x nor f provided.'
+        f = fft(x.double(), dim=-1)
+        device = x.device
+    else:
+        device = f.device
+    n = f.shape[-1]
+    random_phase = _new_random_fft_phase[n % 2](
+        n,
+        device=device,
+        random_state=random_state
+    )
+    f_shifted = f * torch.exp(eps * random_phase)
+    shifted = ifft(f_shifted, dim=-1)
+    return shifted.real.float()
+
+
+def fft_surrogate(X, y, magnitude, random_state, *args, **kwargs):
+    transformed_X = _fft_surrogate(
+        x=X,
+        eps=magnitude,
+        random_state=random_state
+    )
+    return transformed_X, y
+
+
+def _pick_channels_randomly(X, magnitude, random_state):
+    rng = check_random_state(random_state)
+    batch_size, n_channels, _ = X.shape
+    # allows to use the same RNG
+    unif_samples = torch.as_tensor(
+        rng.uniform(0, 1, size=(batch_size, n_channels)),
+        dtype=torch.float,
+        device=X.device,
+    )
+    # equivalent to a 0s and 1s mask, but allows to backprop through
+    # could also be done using torch.distributions.RelaxedBernoulli
+    return torch.sigmoid(1000*(unif_samples - magnitude)).to(X.device)
+
+
+def channel_dropout(X, y, magnitude, random_state, *args, **kwargs):
```

Comment:
all channels which are not private MUST have a docstring and be listed in the public API page.
If it's meant to be private call it _channel_dropout

Reply:
and this is even so for this file `_functional.py` which has underscore in the filename?